src/net.h

```diff
@@ -1023,6 +1052,9 @@ class CConnman
     std::chrono::seconds nMaxOutboundCycleStartTime GUARDED_BY(m_total_bytes_sent_mutex) {0};
     uint64_t nMaxOutboundLimit GUARDED_BY(m_total_bytes_sent_mutex);
 
+    // Stores number of full-tx connections (outbound and manual) per network
+    std::map<Network, std::atomic<unsigned int>> m_network_conn_counts;
```

Comment:
This map can be accessed from multiple threads and has to be protected by a mutex. It is too dangerous to assume that it will never be modified. [operator[]](https://en.cppreference.com/w/cpp/container/map/operator_at) which is used later in the code is not `const`. For example, it is not guaranteed that it will not cause some internal reorg of the structure.

`std::unordered_map` suits better here because we don't need an order and its access and insertion complexity is `O(1)`.

If you really like using unprotected structure of atomics, then maybe `std::array` would be ok (but have to index using `Network` which smells bad). Another option - would it be possible to make this map or array `const`?

Reply:
> change this data struct to an array, add explicit numeric values to the Network enum (for index values)

That seems unnecessary, it is good to rely as less as possible on the actual values of the `Network` enums. That is - the code better not break if `NET_ONION` is not `3`, for example. Use them just as labels. `std::unordered_map` fits best with that - `std::unordered_map<Network, size_t>` would work even if `NET_*` are changed to be strings!

> ... protect the array with m_nodes_mutex to keep the values consistent between the relevant data structures tracking node data

Yes, that sounds like a good idea!

> ... spill liquid on my laptop ...

Oh, good luck recovering! :crossed_fingers: 

> isn't the point to update the values?

Right, that does not make sense. I was thinking about something like this:

```cpp
    const std::map<Network, size_t> m{{NET_IPV4, 0}, {NET_IPV6, 0}, ...};
    ++m[NET_IPV4]; // modify only the value, not the map structure, not possible
```